frontend/src/components/common/Resource/AuthBasedComponent.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { KubeObject } from '../../../lib/k8s/cluster';
+
+export interface AuthBasedComponentProps extends React.PropsWithChildren<{}> {
+  item?: KubeObject;
+  authVerb?: string;
+  subresource?: string;
+  onError?: (err: Error) => void;
+  onUnauthorized?: () => void;
+}
+
+export default function AuthBasedComponent(props: AuthBasedComponentProps) {
```

Comment:
How about `AuthVisible` instead? `AuthBasedComponent` doesn't really spell what it does.

Reply:
I struggled with the component name a bit. I had it as `AuthBasedButton` which makes sense as a name, but it wraps any component, not really just a button, so I renamed it.

I like your `AuthVisible` suggestion and will make the changes.